### ISSUE TITLE
tick with variable rate

### ISSUE THREAD

**Issue by gozzarda** [reflex-frp/reflex-vty#31]
Hi,

I am using reflex-platform and reflex-vty checked out at [`3863af6`](https://github.com/reflex-frp/reflex-vty/commit/3863af6ca018359df5eb16b4bd0281121f61c3e0).

My apologies in advance if this is the wrong place to ask this.

I have been trying to figure how to vary the rate of a tick in my experimental reflex-vty app. After much digging I found reflex-dom's [`tickWithSpeedSelector`](https://github.com/reflex-frp/reflex-examples/blob/42e65be1c74f86413366a32a658c03add9ed0037/frontend/src/Frontend/Examples/ECharts/Main.hs#L75) example, which seems to perfectly fit the bill. Unfortunately, the implementation of `tickWithSpeedSelector` is dependent on `dyn`/`widgetHold`, which I can't seem to find any equivalent for in reflex-vty.

Applying `tickLossyFromPostBuildTime` gets us `(Event t NominalDiffTime) -> (Event t (m (Event t TickInfo)))`, and `switchHold never` would let us `(Event t (Event t TickInfo)) -> m (Event t TickInfo)`, but without `widgetHold` I can't figure out how to plumb these together.

I also think that the general behaviour of `widgetHold` is very useful, and think it would be neat to see a more generic `(Event t (m a)) -> m (Event t a)` implementation if such a thing is possible.

I had a look at how I might go about implementing this myself, but I think I am truly in over my head (I am very new to reflex).

In case it is interesting or useful, I found a vaguely similar problem [here](https://stackoverflow.com/questions/47057131/reflex-delaying-events-based-on-a-behavior).

In summary, I am looking for a way to have a tick at a variable rate that can be updated by another event. This is currently possible in reflex-dom thanks to `widgetHold`, but I am at a loss for how to do it without reflex-dom.

**Comment by gozzarda** [comment]
Of course, as is always the rule, I found the solution immediately after posting about it.

The generalized form of `widgetHold` I wanted already exists as `networkView`/`networkHold` from [Reflex.Network](https://hackage.haskell.org/package/reflex/docs/Reflex-Network.html).

With this, I have been able to assemble the following:
```haskell
tickWithVarRate
  :: (PostBuild t m, PerformEvent t m, TriggerEvent t m, MonadIO (Performable m), MonadFix m, Adjustable t m, NotReady t m, MonadHold t m)
  => (Dynamic t NominalDiffTime) -> m (Event t TickInfo)
tickWithVarRate r = networkView (tickLossyFromPostBuildTime <$> r) >>= switchHold never
```

This, as one might expect, produces an `(Event t TickInfo)` that fires whenever the current timer fires, and replaces the current timer with a new one whenever the given `(Dynamic t NominalDiffTime)` is updated, with the new timer ticking with the given time. Note that of course this means that the timer is restarted as well, so if you keep changing the timing before it has a chance to fire it never will.

Sorry for the pointless issue. Thanks for the neat library.
